api_client/python/timesketch_api_client/search.py

```diff
@@ -0,0 +1,840 @@
+        chip.end_time = end_time
+        self.add_chip(chip)
+
+    @property
+    def chips(self):
+        """Property that returns all the chips in the search object."""
+        return self._chips
+
+    def commit(self):
+        """Commit changes to the search object."""
+        self._raw_response = None
+        super().commit()
+
+    @property
+    def created_at(self):
+        """Property that returns back the creation time of a search."""
+        return self._created_at
+
+    def delete(self):
+        """Deletes the saved search from the store."""
+        if not self._resource_id:
+            logger.warning(
+                'Unable to delete the saved search, it does not appear to be '
+                'saved in the first place.')
+            return False
+
+        resource_url = (
+            f'{self.api.api_root}/sketches/{self._sketch.id}/views/'
+            f'{self._resource_id}/')
+        response = self.api.session.delete(resource_url)
+        return error.check_return_status(response, logger)
+
+    @property
+    def description(self):
+        """Property that returns back the description of the saved search."""
+        return self._description
+
+    @description.setter
+    def description(self, description):
+        """Make changes to the saved search description field."""
+        self._description = description
+        self.commit()
+
+    def from_explore(  # pylint: disable=arguments-differ
+            self,
+            query_string=None,
+            query_dsl=None,
+            query_filter=None,
+            return_fields=None,
+            max_entries=None,
+            **kwargs):
+        """Explore the sketch.
+
+        Args:
+            query_string (str): Elasticsearch query string.
+            query_dsl (str): Elasticsearch query DSL as JSON string.
+            query_filter (dict): Filter for the query as a dict.
+            return_fields (str): A comma separated string with a list of fields
+                that should be included in the response. Optional and defaults
+                to None.
+            max_entries (int): Optional integer denoting a best effort to limit
+                the output size to the number of events. Events are read in,
+                10k at a time so there may be more events in the answer back
+                than this number denotes, this is a best effort.
+            kwargs (dict[str, object]): Depending on the resource they may
+                require different sets of arguments to be able to run a raw
+                API request.
+
+        Raises:
+            ValueError: if unable to query for the results.
+            RuntimeError: if the query is missing needed values, or if the
+                sketch is archived.
+        """
+        super().from_explore(**kwargs)
+        if not (query_string or query_filter or query_dsl):
+            raise RuntimeError('You need to supply a query')
+
+        self._username = getpass.getuser()
```

Comment:
Don't you wanna get the username from the .rc file?

Reply:
hmmm.... so that would probably be better... get the actual logged in user, instead of the system user like we are doing right now.... let me change that.